File: abstract-factory/source-codes/java/src/main/java/com/weekstone/learn/designpattern/abstractfactory/standard/ConcreteFactory1.java
package com.weekstone.learn.designpattern.abstractfactory.standard;

public class ConcreteFactory1 implements Factory {
    @Override
    public ProductA createProductA() {
        return new ConcreteProductA1();
    }

    @Override
    public ProductB createProductB() {
        return new ConcreteProductB1();
    }
}

File: abstract-factory/source-codes/python/app/concrete_factory1.py
from .factory import Factory

from .product_a import ProductA
from .product_b import ProductB
from .product_a1 import ProductA1
from .product_b1 import ProductB1


class ConcreteFactory1(Factory):
    def create_product_a(self) -> ProductA:
        return ProductA1()

    def create_product_b(self) -> ProductB:
        return ProductB1()
